control/controlhttp/client.go

```diff
@@ -424,11 +425,14 @@ func (a *Dialer) tryURLUpgrade(ctx context.Context, u *url.URL, optAddr netip.Ad
 	}
 
 	if optACEHost != "" {
-		dialer = (&ace.Dialer{
-			ACEHost:   optACEHost,
-			ACEHostIP: optAddr, // may be zero
-			NetDialer: dialer,
-		}).Dial
+		if !buildfeatures.HasACE {
+			return nil, feature.ErrUnavailable
+		}
+		f, ok := HookMakeACEDialer.GetOk()
+		if !ok {
+			return nil, feature.ErrUnavailable
+		}
+		dialer = f(dialer, optACEHost, optAddr)
```

Comment:
Documentation (non-blocking): it is too bad that there isn’t a good automatic way to link the ace.mkDialer doc to this use. Consider hardcoding a comment close by that points people towards that initializer. There is only ever one implementation of this, and it would be nice not to have to go splunking.